chaincode/marbles_transfer/marbles_transfer.go

```diff
@@ -361,9 +395,44 @@ func transferMarbleState(ctx contractapi.TransactionContextInterface, marble *Ma
 		return fmt.Errorf("failed to put Marble private properties for buyer: %s", err.Error())
 	}
 
-	// TODO delete the price records for buyer and seller
+	// Delete the price records for seller
+	marblePriceKey, err := ctx.GetStub().CreateCompositeKey(typeMarbleForSale, []string{marble.ID})
+	if err != nil {
+		return fmt.Errorf("failed to create composite key: %s", err.Error())
+	}
 
-	// TODO add a state record for a 'receipt' in both buyer and seller private data collection to record the sales price and date
+	err = ctx.GetStub().DelPrivateData(collectionSeller, marblePriceKey)
+	if err != nil {
+		return fmt.Errorf("failed to delete marble price from implicit private data collection for seller: %s", err.Error())
+	}
+
+	// Delete the price records for buyer
+	marblePriceKey, err = ctx.GetStub().CreateCompositeKey(typeMarbleBid, []string{marble.ID})
+	if err != nil {
+		return fmt.Errorf("failed to create composite key: %s", err.Error())
+	}
+
+	err = ctx.GetStub().DelPrivateData(collectionBuyer, marblePriceKey)
+	if err != nil {
+		return fmt.Errorf("failed to delete marble price from implicit private data collection for buyer: %s", err.Error())
+	}
+
+	// Keep record for a 'receipt' in both buyer and seller private data collection to record the sales price and date
+	// Persist the agreed to price in a collection sub-namespace based on receipt key prefix
+	receipt, err := ctx.GetStub().CreateCompositeKey(typeMarbleReceipt, []string{marble.ID})
```

Comment:
It is possible that the org owns and then sells the same asset twice.
Therefore let's add one more part to the composite key to ensure the receipt is unique.
The txid is probably the best string to add to the composite key to ensure uniqueness. Additionally it will serve as a pointer to the transaction that transferred the asset in case somebody wants to explore the history.
You can get txid using shim's GetTxID().

If you think about it, you'll see that this key approach would enable somebody to query for all asset receipts, or query for the receipts for a specific asset id, depending on the partial composite key query used. Neat huh?